sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -54,14 +54,14 @@
 import shutil
 import sys
 import tempfile
-from distutils.version import StrictVersion
 from typing import Callable
 from typing import List
 from typing import Optional
 from typing import Tuple
 from urllib.parse import urlparse
 
 import pkg_resources
+from pkg_resources import parse_version
```

Comment:
I read more about the integration of distutils with setuptools.  Setuptools has been maintaining a copy of disutils. If we import setuptools, it also imports `setuptools/distutils` or replaces(if already imported) `stdlib/distutils` with `setuptools/distutils`. 

In this PR, i made changes to reflect
1. setuptools to be imported before distutils. [[1](https://github.com/pypa/setuptools/blob/main/_distutils_hack/__init__.py)]
2. Replace the usage of distutils with alternatives. For example, for StricVersion, we can replace it with `pkg_resources.parse_version` since both follow [PEP-440](https://peps.python.org/pep-0440/)
3. Update setuptools while building wheels and building python containers since these images use old version of setuptools

Reply:
Now, its not a strict requirement to have setuptools >= 60.0.0

Reply:
Am I understanding this correctly: Since `setuptools` has its own copy of `distutils`, this means we can still use the original `distutils` logic (provided we make your changes), right?

And you said we can replace `StrictVersion` with the alternative `parse_version`. Besides minimizing usage of `distutils`, are there any other benefits of using `parse_version`? Maybe in the future, when don't want to use `setuptools'` copy of `distutils` anymore, it would be easier to refactor any remaining usage of `distutils`?

Finally, I saw setuptools' [warning](https://github.com/pypa/setuptools/blob/main/_distutils_hack/__init__.py) that "To avoid these issues, avoid "using distutils directly, ensure that setuptools is installed in the traditional way (e.g. not an editable install), and/or make sure that setuptools is always imported before distutils." To me, it feels safer (and easier) to manger to eliminate using `distutils` entirely than to maintain our codebase such that `setuptools` is always imported before `distutils`.

Reply:
We need setuptools to build the `sdist`. I eliminated the usage of distutils almost in the code base except for the `DistutilErrorr` which is used in `setup.py`. To import that error, we just need to import it after importing `setuptools`

For `StrictVersion`, `parse_version` was used because it follows PEP-440 and we need to sort the version of pip and some other modules in our code. `parse_version` comes from `pkg_resources` module.